src/modules/aggregator/aggregator.c

```diff
@@ -662,8 +662,14 @@ static void push_cb (flux_t *h, flux_msg_handler_t *mh,
         aggregate_sink (h, ag);
     rc = 0;
 done:
-    if (flux_respond (h, msg, rc < 0 ? saved_errno : 0, NULL) < 0)
-        flux_log_error (h, "aggregator.push: flux_respond");
+    if (rc < 0) {
+        if (flux_respond_error (h, msg, saved_errno, NULL) < 0)
+            flux_log_error (h, "aggregator.push: flux_respond_error");
+    }
+    else {
+        if (flux_respond (h, msg, NULL) < 0)
+            flux_log_error (h, "aggregator.push: flux_respond");
+    }
 }
```

Comment:
I see a few places where this blob of code is required due to the removal of `errnum` parameter from `flux_respond` -- trading 2 lines of code for 8. The improvement to the function seems like a good idea, but I wonder if we need a convenience macro or function to do it the old way?

You went through and made all the changes, was it overall a net win?

Reply:
I think it was.  As discussed face to face, though, there's a trade off that may be stylistic preference for how request handlers are written.

The style in #2089 is primarily used in older flux code.  Success response and error response take the same path through the code, resulting in better coverage, reduced duplication of any required teardown, and thus fewer LOC.  However, care is needed to avoid assuming errno is zero on success, like the introduction of an rc variable and/or saved_errno.  This style is favored by the old `flux_respond()` footprint.

The other style separates the two paths, with one ending in `flux_resopnd_error()` and one ending in one of the other `flux_respond_*()` variants.   Teardown is duplicated.  One path may not have coverage.  However, IMHO the error path and the success path are more clearly delineated.  This style is favored by the new `flux_respond()` footprint.

Sometimes a little refactoring can be done to reduce the amount of duplicated code in both legs in the latter style.

If it's agreeable generally that the newer style is the better choice, then I should probably tack on a commit to update the older stuff.  I started out that way, but quickly broke something and decided to back off.  Perhaps doing it a little more carefully will yield better results.